mods/ra/maps/infiltration/infiltration.lua

```diff
@@ -0,0 +1,380 @@
+Difficulty = Map.LobbyOption("difficulty")
+
+if Difficulty == "hard" then
+	TimerTicks = DateTime.Minutes(25)
+elseif Difficulty == "normal" then
+	TimerTicks = DateTime.Minutes(28)
+else
+	TimerTicks = DateTime.Minutes(31)
+end
+
+Announcements =
+{
+	{ speech = "TwentyMinutesRemaining", delay = DateTime.Minutes(20) },
+	{ speech = "TenMinutesRemaining", delay = DateTime.Minutes(10) },
+	{ speech = "WarningFiveMinutesRemaining", delay = DateTime.Minutes(5) },
+	{ speech = "WarningFourMinutesRemaining", delay = DateTime.Minutes(4) },
+	{ speech = "WarningThreeMinutesRemaining", delay = DateTime.Minutes(3) },
+	{ speech = "WarningTwoMinutesRemaining", delay = DateTime.Minutes(2) },
+	{ speech = "WarningOneMinuteRemaining", delay = DateTime.Minutes(1) }
+}
+
+TownAttackers = { TownAttacker1, TownAttacker2, TownAttacker3, TownAttacker4, TownAttacker5, TownAttacker6, TownAttacker7 }
+
+PatrolPoints1 = { PatrolPoint11.Location, PatrolPoint12.Location, PatrolPoint13.Location, PatrolPoint14.Location, PatrolPoint15.Location }
+PatrolPoints2 = { PatrolPoint21.Location, PatrolPoint22.Location, PatrolPoint23.Location, PatrolPoint24.Location, PatrolPoint25.Location }
+PatrolPoints3 = { PatrolPoint31.Location, PatrolPoint32.Location, PatrolPoint33.Location, PatrolPoint34.Location }
+PatrolPoints4 = { PatrolPoint41.Location, PatrolPoint42.Location, PatrolPoint43.Location, PatrolPoint44.Location, PatrolPoint45.Location }
+
+Patrol1 = { "e1", "e1", "e1", "e1", "e1" }
+Patrol2 = { "e1", "dog.patrol", "dog.patrol" }
+Patrol3 = { "e1", "e1", "dog.patrol" }
+
+TransportType = "lst.unselectable.unloadonly"
+
+SecureLabFailed = function()
+	Utils.Do(humans, function(player)
+		if player then
+			player.MarkFailedObjective(secureLab)
+		end
+	end)
+end
+
+timerStarted = false
+StartTimer = function()
+	Utils.Do(humans, function(player)
+		if player.IsLocalPlayer then
+			TimerColor = player.Color
+		end
+	end)
+	CountDownTimerAnnouncements()
+	ticked = TimerTicks
+	timerStarted = true
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		Utils.Do(humans, function(player)
+			Media.PlaySpeechNotification(player, "TimerStarted")
+		end)
+	end)
+end
+
+CountDownTimerAnnouncements = function()
+	for i = #Announcements, 1, -1 do
+		local delay = TimerTicks - Announcements[i].delay
+		Trigger.AfterDelay(delay, function()
+			if not labSecured then
+				Utils.Do(humans, function(player)
+					Media.PlaySpeechNotification(player, Announcements[i].speech)
+				end)
+			end
+		end)
+	end
+end
+
+reinforcementsHaveArrived = false
+LabInfiltrated = function()
+	Utils.Do(humans, function(player)
+		if player then
+			secureLab = player.AddPrimaryObjective("Secure the laboratory before the soviets can react.")
```

Comment:
This objective is confusing. I wasn't sure what I was meant to do (capture it?, or infiltrate it again?) until I read the script. How about "Destroy all units guarding the lab."?